packages/zoe/src/contracts/vpool-xyk-amm/pool.js

```diff
@@ -0,0 +1,255 @@
+// @ts-check
+
+import { E } from '@agoric/eventual-send';
+import { assert, details as X } from '@agoric/assert';
+import { AssetKind, AmountMath, isNatValue } from '@agoric/ertp';
+import { makeNotifierKit } from '@agoric/notifier';
+import { Far } from '@agoric/marshal';
+
+import {
+  calcLiqValueToMint,
+  calcValueToRemove,
+  calcSecondaryRequired,
+} from '../../contractSupport/index.js';
+
+import '../../../exported.js';
+import { makePriceAuthority } from '../multipoolAutoswap/priceAuthority.js';
+import { makeSinglePool } from './singlePool';
+
+// Pools represent a single pool of liquidity. Price calculations and trading
+// happen in a wrapper class that knows whether the proposed trade involves a
+// single pool or multiple hops.
+
+/**
+ * @param {ContractFacet} zcf
+ * @param {(brand: Brand) => boolean} isSecondary
```

Comment:
`isSecondary` suggests that it's an intrinsic property of the `Brand` rather than a relationship between the current state of the AMM and the `Brand`.

I see that it's defined as `const isSecondary = secondaryBrandToPool.has;`. Perhaps `secondariesHas`?

Reply:
I went with `isInSecondaries`.